src/Form.jsx

```diff
@@ -161,6 +161,10 @@ export default class Form extends React.Component {
   updateField = ({ fieldPath, fieldProps: directFieldProps, propsPatch = null }) => {
     const { fields } = this.state;
     const fieldProps = directFieldProps || fields.getIn([fieldPath]);
+
+    /* Do not update field if its not registered yet */
+    if (!fieldProps) return;
```

Comment:
Could you please describe me why does this happen?

When updateField is called in this bug scenario? Which lifecycle event triggers that? How is that related to initialValue?

How come field is not yet registered?

Reply:
Also please take a look at each event handler on Form level - handleFieldChange, etc. 

All event handlers are bypassed before the field is actually registered. So this issue is somewhat unclear to me. I would suggest more investigation.

I want to look into this once I have some spare time.